### ISSUE TITLE
Filter operation doesn't work if variable name intersects with column name

### ISSUE THREAD

**Issue by mattsgithub** [tidyverse/tidyverse#199]
**TL;DR:**
Using variable names that conflict with dataframe column names in a filter operation, doesn't yield expected results. 

**Example:**
I have a dataframe with a column named `week` which is a datatype of `<date>`. I run a filter such as:

`df_subset <- df %>% filter(week == "2019-08-26")`

Works just fine. Subsets data to those observations. However, if I run:

```
week <- "2019-08-26"
df_subset <- df %>% filter(week == week)
```

it does not subset at all. If I change my variable name to say, `w` it then works.

If it's not possible to permit the use of variable names that intersect with colum names in the tibble, I'd recommend at minimum, logging a warning.

**Comment by jennybc** [comment]
This issue would be better filed in the repository for the dplyr package, which is where `filter()` comes from. This repository is for the tidyverse meta-package, which really doesn't hold any "smarts" *per se*.

But to answer your question, you can use the `{{ }}` "embracing" or "curly curly" syntax in this case:

``` r
library(tidyverse)

(df <- tibble(
  x = 1:3,
  y = letters[x]
))
#> # A tibble: 3 x 2
#>       x y    
#>   <int> <chr>
#> 1     1 a    
#> 2     2 b    
#> 3     3 c

df %>% filter(y == "b")
#> # A tibble: 1 x 2
#>       x y    
#>   <int> <chr>
#> 1     2 b

z <- "b"
df %>% filter(y == z)
#> # A tibble: 1 x 2
#>       x y    
#>   <int> <chr>
#> 1     2 b
df %>% filter(y == {{ z }})
#> # A tibble: 1 x 2
#>       x y    
#>   <int> <chr>
#> 1     2 b

y <- "b"
df %>% filter(y == {{ y }})
#> # A tibble: 1 x 2
#>       x y    
#>   <int> <chr>
#> 1     2 b
```

<sup>Created on 2019-10-04 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9000)</sup>

**Comment by mattsgithub** [comment]
Thanks for the quick response. I did not know about the `{{ }}` notation. Good to know! I will migrate this issue over to `dplyr`. Thanks again.

**Comment by mattsgithub** [comment]
For reference, I've reposted this at https://github.com/tidyverse/dplyr/issues/4578
